static/app/components/core/link/link.tsx

```diff
@@ -67,21 +70,64 @@ const Anchor = styled('a', {
   ${getLinkStyles}
 `;
 
+const preload = async (to: To) => {
+  // Try to match the route and preload if it has a preload method
+  try {
+    const routeConfig = (await import('sentry/routes')).routes();
+    const matches = matchRoutes(routeConfig, to);
+
+    if (matches && matches.length > 0) {
+      // Preload all matching routes, not just the last one
+      for (const match of matches) {
+        const routeHandle = match.route.handle;
+
+        // Check if the handle has a preload method
+        if (
+          routeHandle &&
+          typeof routeHandle === 'object' &&
+          PRELOAD_HANDLE in routeHandle
+        ) {
+          routeHandle[PRELOAD_HANDLE]?.().catch(() => {
+            // Ignore preload errors
+          });
+        }
+      }
+    }
+  } catch {
+    // Silently fail if route matching fails
```

Comment:
Lets add a sentry logger call to the route reject handler and the catch statement here